bin/i18n/resources/pegasus/hourofcode/sync_out.rb

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env ruby
+
+require 'fileutils'
+require 'yaml'
+require 'tempfile'
+
+require_relative '../../../../../dashboard/config/environment'
+require_relative '../../../../../pegasus/helpers/pegasus_languages'
+require_relative '../../../i18n_script_utils'
+require_relative '../hourofcode'
+
+module I18n
+  module Resources
+    module Pegasus
+      module HourOfCode
+        class SyncOut
+          def self.perform
+            new.execute
+          end
+
+          def execute
+            progress_bar.start
+
+            I18nScriptUtils.process_in_threads(pegasus_languages) do |pegasus_lang|
+              crowdin_locale = pegasus_lang[:crowdin_name_s]
+              crowdin_locale_dir = I18nScriptUtils.locale_dir(crowdin_locale, DIR_NAME)
+              next unless File.directory?(crowdin_locale_dir)
```

Comment:
If the directory has already been renamed from for example Spanish to es-ES, the distribution will be skipped.

Reply:
When I ran the full sync, in this step, the `File.directory?(crowdin_locale_dir)` always returned false.

Reply:
It's because you don't have a Crowdin locale dir, e.g. `i18n/locales/Spanish/houofcode`.

The algorithm was optimized:

#### Before:
1. Moves files from `i18n/locales/Spanish/hourofcode` to `i18n/locales/es-ES/hourofcode`
2. Checks if a file in `i18n/locales/es-ES/hourofcode` was changed
3. Distributes the localizations from the `i18n/locales/es-ES/hourofcode` file

#### Now:
1. Distributes the localizations from the `i18n/locales/Spanish/hourofcode` file - we don't need to check if files in the dir were changed since we download only changed files from Crowdin
2. Moves files from `i18n/locales/Spanish/hourofcode` to `i18n/locales/es-ES/hourofcode`